### PR TITLE
Feat/Add notary notification for netmap contract

### LINKED ISSUE #750: Fix bootstrap process without side chain notifications
Alphabet nodes approve fixed version of `netmap.AddPeer` invocation from storage nodes: they add location attributes based on `UN-LOCODE` attribute. This contradicts the way Alphabet nodes will approve notary invocations after https://github.com/nspcc-dev/neofs-node/issues/519. We should figure out the fix for this.

Possible solutions
1. keep using notifications for `AddPeer` invocations with 1KiB limit of NodeInfo structure,
2. do not sign notary transaction from Storage node and sign new transactions built manually by Alphabet nodes,
3. parse `UN-LOCODE` attribute on Storage node (requires valid copy of LOCODE database on every storage node or store it side chain).

### LINKED ISSUE #805: Use notary notifications in netmap contract invocations
Related to https://github.com/nspcc-dev/neofs-node/pull/770

- [ ] `AddPeer` method
- [ ] `UpdateState` method

AddPeer has an issue: https://github.com/nspcc-dev/neofs-node/issues/750

### PR DESCRIPTION
**Pull request by carpawell**
Closes #805 and closes #750.

### CODE REVIEW

File: pkg/morph/event/netmap/update_peer_notary.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+		switch {
+		case fieldNum == 0 && opcode.PUSHDATA1 <= currCode && currCode <= opcode.PUSHDATA4:
+			err = ev.setPublicKey(op.Param())
+			if err != nil {
+				return nil, err
+			}
+
+			fieldNum++
+		case fieldNum == 1 && opcode.PUSH0 <= currCode && currCode <= opcode.PUSH16:
```
Comment (alexvanin) [inline pkg/morph/event/netmap/update_peer_notary.go:58]: This should fail on the number bigger than 16. There are only couple of states, yet I propose to process this case as well.

File: pkg/morph/event/netmap/update_peer_notary.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	"github.com/nspcc-dev/neo-go/pkg/crypto/keys"
+	"github.com/nspcc-dev/neo-go/pkg/vm/opcode"
+	"github.com/nspcc-dev/neofs-api-go/pkg/netmap"
+	netmapv2 "github.com/nspcc-dev/neofs-api-go/v2/netmap"
+	"github.com/nspcc-dev/neofs-node/pkg/morph/event"
+)
+
+func (s *UpdatePeer) setPublicKey(v []byte) (err error) {
+	if v != nil {
```
Comment (cthulhu-rider) [inline pkg/morph/event/netmap/update_peer_notary.go:15]: Is `v == nil` case correct?

File: pkg/morph/event/netmap/add_peer_notary.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+		switch {
+		case opcode.PUSHDATA1 <= currentOp && currentOp <= opcode.PUSHDATA4:
+			if fieldNum == expectedItemNumAddPeer {
+				return nil, event.UnexpectedArgNumErr(AddPeerNotaryEvent)
+			}
+
+			ev.setNode(op.Param())
+			fieldNum++
+		case opcode.PUSH0 <= currentOp && currentOp <= opcode.PUSH16 || currentOp == opcode.PACK:
```
Comment (alexvanin) [inline pkg/morph/event/netmap/add_peer_notary.go:41]: Rebase and remove this case

File: pkg/morph/event/netmap/update_peer_notary.go (hunk 1 above)
Comment (carpawell) [reply pkg/morph/event/netmap/update_peer_notary.go:58]: added parsing general `INT` here

File: pkg/morph/event/netmap/update_peer_notary.go (hunk 2 above)
Comment (carpawell) [reply pkg/morph/event/netmap/update_peer_notary.go:15]: in fact, it is impossible to get nil slice from `neo-go` library here, but added check for it

File: pkg/morph/event/netmap/add_peer_notary.go (hunk 3 above)
Comment (carpawell) [reply pkg/morph/event/netmap/add_peer_notary.go:41]: rebased and deleted that case

**Review by alexvanin — APPROVED** [review]

**Pull request merged**